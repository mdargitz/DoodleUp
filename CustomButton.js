import React from 'react';
import { TouchableOpacity, Text, View } from 'react-native';

class CustomButton extends React.Component {
  render() {
    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity 
          style={styles.touchStyle}
          onPress={this.props.onPress}>
          <Text style={styles.textStyle}>{this.props.buttonText}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = {
  viewStyle : {
    alignItems: 'center'
  },
  touchStyle : {
    margin: 20,
    padding: 10,
    backgroundColor: '#bc7c9c',
    borderRadius: 5,

  },
  textStyle : {
    color: '#fff',
    fontSize: 30,
  }
};

export default CustomButton;