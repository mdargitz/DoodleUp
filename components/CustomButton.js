import React from 'react';
import { TouchableOpacity, Text, View } from 'react-native';

class CustomButton extends React.Component {
  render() {
    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity 
          style={styles.touchStyle}
          onPress={this.props.onPress}>
          <Text style={styles.textStyle}>{this.props.buttonText}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = {
  viewStyle : {
    alignItems: 'center'
  },
  touchStyle : {
    margin: 8,
    padding: 10,
    backgroundColor: '#bc7c9c',
    borderRadius: 5,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
  textStyle : {
    color: '#fff',
    fontSize: 30,
  }
};

export default CustomButton;