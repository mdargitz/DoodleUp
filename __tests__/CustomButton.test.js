import React from 'react';
import CustomButton from '../components/CustomButton';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<CustomButton />).toJSON();
  expect(tree).toMatchSnapshot();
});