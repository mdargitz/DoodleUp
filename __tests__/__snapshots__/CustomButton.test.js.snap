// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#bc7c9c",
        "borderRadius": 5,
        "margin": 8,
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    />
  </View>
</View>
`;
